From b791985ae4c49608632100b3f0bcac8832507a81 Mon Sep 17 00:00:00 2001
From: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Date: Tue, 8 Mar 2016 09:29:37 -0500
Subject: [PATCH 1489/1565] drm/amd/dal: Fix variadic register update/set
 function.

Avoid using var definition in a macro to avoid compile errors.

Signed-off-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dm_services.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dm_services.h b/drivers/gpu/drm/amd/dal/dc/dm_services.h
index ffae50f..15417e4 100644
--- a/drivers/gpu/drm/amd/dal/dc/dm_services.h
+++ b/drivers/gpu/drm/amd/dal/dc/dm_services.h
@@ -170,9 +170,8 @@ static inline void generic_reg_update_ex(const struct dc_context *ctx,
 }
 
 #define generic_reg_update(ctx, inst_offset, reg_name, n, ...)\
-		uint32_t reg_val = dm_read_reg(ctx, mm##reg_name + inst_offset);	\
 		generic_reg_update_ex(ctx, \
-		mm##reg_name + inst_offset, reg_val, n, \
+		mm##reg_name + inst_offset, dm_read_reg(ctx, mm##reg_name + inst_offset), n, \
 		__VA_ARGS__)
 
 #define generic_reg_set(ctx, inst_offset, reg_name, n, ...)\
-- 
1.9.1

